web/src/message_edit.ts

```diff
@@ -82,6 +83,24 @@ type EchoedMessageData = {
     mentioned_me_directly: boolean;
 };
 
+const detached_files_api_response_schema = z.object({
+    detached_files: z.array(
+        z.object({
+            id: z.number(),
+            name: z.string(),
+            path_id: z.string(),
+            size: z.number(),
+            create_time: z.number(),
+            messages: z.array(
+                z.object({
+                    id: z.number(),
+                    date_sent: z.number(),
+                }),
+            ),
+        }),
+    ),
+});
```

Comment:
Can this schema be shared with `attachment_api_response_schema`? It'd be nice to not duplicate it.

Possibly this entails making a new `attachments.ts` that just includes the base attachment zod schema that can be used by both callers.

Reply:
Yeah okay, took me a while to figure out how to reuse the schemas. Good idea, I didn't think about this at all.

Reply:
Added a new prep commit for this.